kong/router/init.lua

```diff
@@ -10,6 +10,117 @@ local atc_compat = require("kong.router.atc_compat")
 local utils = require("kong.router.utils")
 local is_http = ngx.config.subsystem == "http"
 
+local DEFAULT_MINIMAL_INTERVAL = 60 -- 1 minute
+
+local phonehome_statistics
+local phonehome_sync_start
+do
+  local reports = require("kong.reports")
+  local semaphore = require("ngx.semaphore")
+  local nkeys = require("table.nkeys")
+  local empty_table = {}
+  -- reuse tables to avoid cost of creating tables and garbage collection
+  local protocol_counts = {
+    http = 0, -- { "http", "https" },
+    stream = 0, -- { "tcp", "tls", "udp" },
+    tls_passthrough = 0, -- { "tls_passthrough" },
+    grpc = 0, -- { "grpc", "grpcs" },
+  }
+  local path_handling_counts = {
+    v0 = 0,
+    v1 = 0,
+  }
+  local route_report = {
+    flavor = "unknown",
+    paths_count = 0,
+    headers_count = 0,
+    total = 0,
+    regex_paths_count = 0,
+    protocols = protocol_counts,
+    path_handling = path_handling_counts,
+  }
+
+
+  local need_report = assert(semaphore.new())
+
+  local function report_timer_impl(premature)
+    if premature then
+      return
+    end
+
+    if need_report:wait(DEFAULT_MINIMAL_INTERVAL) then
+      reports.send("routes", route_report)
```

Comment:
This is over complicating things, please remove all timers and semaphore from this.

We trigger a phone home for each router rebuild, please avoid making this more complicated than necessary for such a non-important feature that goes into patch releases.

We should simple follow how https://github.com/Kong/kong/blob/master/kong/api/routes/config.lua#L23 is done by sending a message when router is rebuild, and avoid bring in more complexity than necessary.